src/util/error-extractor/index.js

```diff
@@ -0,0 +1,68 @@
+
+class ErrorDetailsExtractor {
+  constructor(extractObjBuilder) {
+    this.status=extractObjBuilder.status;
+    this.messageDetails = { ...extractObjBuilder.messageDetails }
+  }
+
+  static get Builder() {
+    return ErrorDetailsExtractBuilder;
+  }
+  
+}
+
+
+class ErrorDetailsExtractBuilder {
```

Comment:
is a builder required for this? only 2 params right?

Reply:
The thought process behind this is to make sure that tomorrow if we need to add more fields(for error enrichment) like may be actionItem or something else, it remains easy to initialise without making any changes to the error detail extractor in general.

Reply:
Also if you see, we are having two ways in which `message` can be set(one way in which a field in error-response which contains the error message or provide the error message itself(by dev in source-code)), I have also taken this into consideration while going for the builder